spec/Appendix B -- Grammar Summary.md

```diff
@@ -176,6 +180,124 @@ NonNullType :
   - NamedType !
   - ListType !
 
-Directives : Directive+
+Directives[Const] : Directive[?Const]+
+
+Directive[Const] : @ Name Arguments[?Const]?
+
+TypeSystemDefinition :
+  - SchemaDefinition
+  - TypeDefinition
+  - TypeExtension
+  - DirectiveDefinition
+
+SchemaDefinition : schema Directives[Const]? { OperationTypeDefinition+ }
+
+OperationTypeDefinition : OperationType : NamedType
+
+Description : StringValue
+
+TypeDefinition :
+  - ScalarTypeDefinition
+  - ObjectTypeDefinition
+  - InterfaceTypeDefinition
+  - UnionTypeDefinition
+  - EnumTypeDefinition
+  - InputObjectTypeDefinition
+
+TypeExtension :
+  - ScalarTypeExtension
+  - ObjectTypeExtension
+  - InterfaceTypeExtension
+  - UnionTypeExtension
+  - EnumTypeExtension
+  - InputObjectTypeExtension
+
+ScalarTypeDefinition : Description? scalar Name Directives[Const]?
+
+ScalarTypeExtension :
+  - extend scalar Name Directives[Const]
+
+ObjectTypeDefinition : Description? type Name ImplementsInterfaces? Directives[Const]? FieldsDefinition?
+
+ObjectTypeExtension :
+  - extend type Name ImplementsInterfaces? Directives[Const]? FieldsDefinition
+  - extend type Name ImplementsInterfaces? Directives[Const]
+  - extend type Name ImplementsInterfaces
+
+ImplementsInterfaces :
+  - implements `&`? NamedType
+  - ImplementsInterfaces & NamedType
```

Comment:
is it for this case (aka formatting??)

    type Foo implements
        & Bar 
        & Baz 
    {
       ...

?

Reply:
Again it might be I am reading the grammar decl wrong but currently I think it allows

    type Foo implements & Bar implements & Baz

(because of the repeating ??)

Reply:
@bbakerman Yes leading `&` supported for formatting more context here: https://github.com/facebook/graphql/pull/323

> Again it might be I am reading the grammar decl wrong but currently I think it allows

No, it doesn't. According to the grammar `implements` can be used only in the begging.